packages/react-app/src/hooks/useStaticJsonRPC.js

```diff
@@ -25,9 +25,23 @@ export default function useStaticJsonRPC(urlArray) {
   }, [urlArray]);
 
   useEffect(() => {
-    handleProviders();
+    //  localProvider is null  and urlArray's length is 1 then load a first  localProvider
+    if (localProvider === null && urlArray.length === 1) {
+      handleProviders();
+    }
+
+    //  localProvider is not  null check chain id if it is mainnet id  then set provider
+    if (localProvider !== null && localProvider?._network.chainId === 1) {
+      setProvider(provider);
```

Comment:
You are setting an empty provider here.

It should be `setProvider(localProvider)`.

Also the logic on this useEffect is a bit confusing. Not sure if we need those conditions. I think it could be something like:

```js
    if (localProvider && localProvider?._network?.chainId === 1) {
      setProvider(localProvider);
      return;
    }

    handleProviders();
```

Reply:
ohh sry its a type :)